app/models/laa_reference.rb

```diff
@@ -5,8 +5,8 @@ class LaaReference < ApplicationRecord
   validates :maat_reference, presence: true, uniqueness: { conditions: -> { where(linked: true) } }
   validates :user_name, presence: true
 
```

Comment:
It may be worth to add validation for `unlink_reason_code`, for example: 
`validates :unlink_reason_code, numericality: { only_integer: true, greater_than_or_equal_to: 0, less_than_or_equal_to: 100 }`